packages/eslint-plugin/lib/rules/callable-types.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @fileoverview Use function types instead of interfaces with call signatures
+ * @author Benjamin Lichtman
+ */
+'use strict';
+const util = require('../util');
+
+/**
+ * @typedef {import("eslint").Rule.RuleModule} RuleModule
+ * @typedef {import("estree").Node} ESTreeNode
+ */
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * @type {RuleModule}
+ */
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Use function types instead of interfaces with call signatures',
+      category: 'TypeScript',
+      recommended: false,
+      extraDescription: [util.tslintRule('callable-types')],
+      url: util.metaDocsUrl('callable-types')
+    },
+    fixable: 'code',
+    messages: {
+      callableTypeViolation:
+        "{{ type }} has only a call signature - use '{{ sigSuggestion }}' instead."
+    },
+    schema: [],
+    type: 'suggestion'
+  },
+
+  create(context) {
+    const sourceCode = context.getSourceCode();
+
+    //----------------------------------------------------------------------
+    // Helpers
+    //----------------------------------------------------------------------
+
+    /**
+     * Checks if there is no supertype or if the supertype is 'Function'
+     * @param {ESTreeNode} node The node being checked
+     * @returns {boolean} Returns true iff there is no supertype or if the supertype is 'Function'
+     */
+    function noSupertype(node) {
+      if (
+        typeof node.extends === 'undefined' &&
+        typeof node.implements === 'undefined'
```

Comment:
better to check this as a boolean cast so it handles null case
```suggestion
        !node.extends && !node.implements
```

Reply:
you should check only `!node.extends`

Reply:
Changed to just check `extends`.